### ISSUE TITLE
Linux and Mac Support

### ISSUE THREAD

**Issue by devloop0** [the-machine-project/the-machine#23]
Currently, our machine supports Windows (only platform I could test on as of now), but porting everything to Linux and Mac should be easy. I would definitely appreciate some community support on this. Basically, the only platform-dependent line of code is in MainPage.java with the System.loadLibrary.

**Comment by ErnestThornhill** [comment]
This [website](http://dummyscodes.blogspot.com/2015/12/develop-opencv-cross-platform.html) shows how cross OS works with openCV, it should be proably be the same even though you guys are using JavaCV. For an example the [LoadLibrary](https://github.com/ErnestThornhill/The-Machine/blob/master/Java%20Code/The%20Machine/src/theMachine/Webcam/LoadLibrary.java) class in my github shows how the cross-os would work. Also in that github are the [Linux](https://github.com/ErnestThornhill/The-Machine/tree/master/Java%20Code/The%20Machine/resources/theMachine/opencv/linux) and [Mac](https://github.com/ErnestThornhill/The-Machine/tree/master/Java%20Code/The%20Machine/resources/theMachine/opencv/mac) opencv files that are necessary for the program to work in the other operating systems (I have not had a chance to test them in their said operating system but I believe that they should work well). By the way if using this method for cross-os, you'll need to replace the System.loadLibrary with a call to LoadLibrary.loadOpenCV() in a static block before you initialize anything. Hopefully this works!

**Comment by devloop0** [comment]
Thanks, I will check it out and close this issue as soon as I get a prototype working and test it.

**Comment by xnart** [comment]
There is should be more easier way. 

`System.loadLibrary( Core.NATIVE_LIBRARY_NAME );`

I believe this code sees what is your platform, and produces right file name for that. If you put all platform native libraries in your libraries path, it should grab for you.

I didn't try but it has to be something like that.

**Comment by xnart** [comment]
I built native library on my Linux x64. Here is the library file : 

[opencv_java310.so.zip](https://github.com/the-machine-project/the-machine/files/282381/opencv_java310.so.zip)

**Comment by devloop0** [comment]
What architectures does it support? Also, I will try to integrate these for a Linux build as soon as possible.  Thank you for doing this.

**Comment by xnart** [comment]
Since i built it on x64 architecture, i think it supports only x64s. But not sure. 

I tried this library file with The Machine and it works on my Arch Linux x64.

**Comment by devloop0** [comment]
Okay, I have an x64 installation as well so I will test it on my local machine.
Also, did you hardcode the opencv_java310.so path in MainPage.java?

**Comment by xnart** [comment]
Yeah, i couldn't run it with System.loadLibrary(). It runs with : System.load(ABSOLUTE PATH .so). 

Got to search out how Maven deals with native libraries.

**Comment by 8bitgentleman** [comment]
If a mac is needed to compile an OSX version I can lend a hand. Just tell me what to do.

**Comment by devloop0** [comment]
If you manage to compile everything from source on the latest master branch with the correct OpenCV dependency and modify the UserPlatform.java file to load that file on a Mac, that should be enough.

**Comment by RoccoDeveloping** [comment]
Can you post the UserPlatform.java for Macs please? I have a huge Thunderbolt Display and it can only be attached with Macs.
